src/mytardis_client/endpoint_info.py

```diff
@@ -0,0 +1,179 @@
+"""Definitions of the properties of MyTardis endpoints"""
+
+from typing import Optional
+
+from pydantic import BaseModel
+
+from src.mytardis_client.endpoints import MyTardisEndpoint
+from src.mytardis_client.objects import MyTardisObject
+
+
+class GetRequestProperties(BaseModel):
+    """Definition of behaviour/structure for a GET request to a MyTardis endpoint."""
+
+    # Note: it would be useful here to store the dataclass type for the response, so that
```

Comment:
Is it worth considering using dependency injection here? We could __init__ the class with a dataclass type as an optional object. That would maintain the self-containment of this module while also giving us the option of injecting a dataclass. We'd then need a check for None-ness with validation behaviour if the dataclass was provided.

Reply:
I'm doing some more work in this area as a follow-on from this PR, so I'll keep this in mind and maybe we can pick up the discussion once that's ready?